### PR TITLE
Test program that spawns and ends threads.

### PR DESCRIPTION
**Pull request by danielfenner**
Helpful to test the behaviour when attaching to a process that races with us by creating tasks.

### CODE REVIEW

File: OrbitTest/main_short_lived_threads.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+// Main thread goes into a busy loop that spawns and joins threads.
+// The spawned threads go into a busy wait.
+// The first command line parameter gives the number of threads to spawn.
+// The second command line parameter gives the live time of each spawned thread.
+// The actual live time is varied between 100% and 200% of that value to
+// make things slightly more interesting.
+int main(int argc, char* argv[]) {
+  if (argc != 3) {
+    std::cerr << "Usage:\n./OrbitTestShortLivedThreads number_of_threads time_to_live_per_thread\n";
```
Comment (pierricgimmig) [inline OrbitTest/main_short_lived_threads.cpp:37]: Maybe specify the time unit: time_to_live_per_thread_ms?

File: OrbitTest/main_short_lived_threads.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// Busy wait for ttl_ms.
+void Worker(int ttl_ms) {
+  auto deadline = std::chrono::system_clock::now() + std::chrono::milliseconds(ttl_ms);
+  while (true) {
+    if (std::chrono::system_clock::now() > deadline) {
+      break;
+    }
+  }
+  std::lock_guard lock(g_joinable_mutex);
+  g_joinable_threads.emplace(std::this_thread::get_id());
```
Comment (pierricgimmig) [inline OrbitTest/main_short_lived_threads.cpp:26]: Is the g_joinable_threads set really needed? We could simply create all threads and then join them all, no?

Comment (danielfenner) [reply OrbitTest/main_short_lived_threads.cpp:26]: I'd like to keep _number_of_threads_ threads running all the time to keep the system under stress. So in every round in the main loop I just want to join the threads that already finished their "work".

File: OrbitTest/main_short_lived_threads.cpp (hunk 1 above)
Comment (danielfenner) [reply OrbitTest/main_short_lived_threads.cpp]: yep, done.

File: OrbitTest/main_short_lived_threads.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// Main thread goes into a busy loop that spawns and joins threads.
+// The spawned threads perform a busy wait.
+// The first command line parameter gives the number of threads to spawn. When
+// a thread finishs it will be replaced by a new one.
```
Comment (pierricgimmig) [inline OrbitTest/main_short_lived_threads.cpp:32]: nit: finishes

File: OrbitTest/main_short_lived_threads.cpp (hunk 2 above)
Comment (pierricgimmig) [reply OrbitTest/main_short_lived_threads.cpp:26]: Ah ok ok. And the lifecycle of threads is what you are interested in? A comment on how this program will be used to add more context could be useful.

**Review by pierricgimmig — APPROVED** [review]

File: OrbitTest/main_short_lived_threads.cpp (hunk 3 above)
Comment (danielfenner) [reply OrbitTest/main_short_lived_threads.cpp]: done

File: OrbitTest/main_short_lived_threads.cpp (hunk 2 above)
Comment (danielfenner) [reply OrbitTest/main_short_lived_threads.cpp:26]: Yes, I'd like to provoke errors in places where we need to keep track of the set of threads in the target process.

Added a few lines of comments.

**Pull request merged**